cmd/root.go

```diff
@@ -45,6 +46,17 @@ func NewRootCmd(out io.Writer, args []string) *cobra.Command {
 	flags.Parse(args)
 	settings = new(EnvSettings)
 
+	// Helm plugins are not passed Helm's global flags but instead must
+	// read the global flag values from each corresponding environment variable.
+	// The below allows to use Helm's --kube-context global flag.
+	if ctx := os.Getenv("HELM_KUBECONTEXT"); ctx != "" {
```

Comment:
Should we check first if  `settings.KubeContext` has a value set? I am asking because if run with the binary (not as a plugin), then we would want `settings.KubeContext` to be used instead of `$HELM_KUBECONTEXT`

Reply:
I've traced this and when the `--kube-context` flag is set when run with the binary, `settings.KubeContext` is set later on from within the `migrateCmd.Execute()` call in `main.go`.  Therefore it overwrites the value of the `HELM_KUBECONTEXT` variable, as desired.

Thankfully, when called as a plugin, `settings.KubeContext` retains its value during the `migrateCmd.Execute()` because its default value is to be set to itself. See https://github.com/helm/helm-2to3/blob/e898519fe62720e51188bed8a6524e739e6d7ebe/cmd/environment.go#L45
It's like you had already prepared for this case!?!

Reply:
By design of course! :)